app/models/user.rb

```diff
@@ -82,4 +85,20 @@ def save_settings!(data)
   def unauthenticated_message
     override_paranoid_setting(false) { super }
   end
+
+  def soft_delete
+    update(deleted_at: Time.zone.now)
+  end
+
+  def active?
+    self.deleted_at.nil?
+  end
+
+  def active_for_authentication?
+    super && active?
+  end
+
+  def inactive_message
+    active? ? super : 'This account has been deleted.'
```

Comment:
We can move this string to devise.en.yml and replace it with a symbol, i.e. :account_deleted